### PR TITLE
[CHIA-1681] Replace black with Ruff

### PR DESCRIPTION
**Pull request by Quexington**
See PR https://github.com/Chia-Network/chia-blockchain/pull/18759 for more information about Ruff. This PR continues the work of simplifying our pre-commit workflow by replacing another one of our tools with Ruff.

Ruff specifically targets black to replace with its formatter since black is so popular in the python ecosystem.  There are, however, some differences in opinion and they are all documented here: https://docs.astral.sh/ruff/formatter/black/ .  The main one that may be a little controversial is the fact that ruff would rather avoid parentheses than break up call expressions like so:
```
derivation_record: Optional[
    DerivationRecord
] = await wallet_state_manager.puzzle_store.get_derivation_record_for_puzzle_hash(coin.puzzle_hash)
```

### CODE REVIEW

File: chia/_tests/util/build_network_protocol_files.py [hunk 1]
```diff
@@ -293,9 +293,8 @@ def main() -> None:
     for name, function in name_to_function.items():
         path = tests_dir.joinpath(name)
         path.write_text(function())
-        # black seems to have trouble when run as a module so not using `python -m black`
         subprocess.run(
-            [scripts_path.joinpath("black"), os.fspath(path.relative_to(tests_dir))],
+            [scripts_path.joinpath("ruff"), "format", os.fspath(path.relative_to(tests_dir))],
```
Comment (altendky) [inline chia/_tests/util/build_network_protocol_files.py:297]: this is what the comment was suggesting i preferred.  or at least i'm guessing that was me.  if it works it is a bit less tied to os behavior and more to python behavior.

```suggestion
            [sys.executable, "-m", "ruff", "format", os.fspath(path.relative_to(tests_dir))],
```

File: chia/_tests/wallet/test_wallet_coin_store.py [hunk 2]
```diff
@@ -787,7 +787,7 @@ async def test_get_coin_records_total_count_cache() -> None:
         assert (await store.get_coin_records(include_total_count=True)).total_count == 3
         # Make sure the total count increases for the same query when rolling back
         assert (await store.get_coin_records(include_total_count=True)).total_count == 3
-        await store.rollback_to_block(0),
+        (await store.rollback_to_block(0),)
```
Comment (altendky) [inline chia/_tests/wallet/test_wallet_coin_store.py:790]: i think this is pointing out an irrelevant 'bug'.  maybe we just stop making the tuple?  let me know if you want this in a separate pr and i'll make that.

```suggestion
        await store.rollback_to_block(0)
```

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/_tests/util/build_network_protocol_files.py (hunk 1 above)
Comment (Quexington) [reply chia/_tests/util/build_network_protocol_files.py]: Addressed in an accidentally poorly named commit

File: chia/_tests/wallet/test_wallet_coin_store.py (hunk 2 above)
Comment (Quexington) [reply chia/_tests/wallet/test_wallet_coin_store.py]: Addressed in an accidentally poorly named commit

**Review by altendky — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**